app/services/moves/finder.rb

```diff
@@ -2,15 +2,40 @@
 
 module Moves
   class Finder
-    attr_accessor :filter_params, :ability
+    attr_reader :filter_params, :ability
 
-    def initialize(filter_params, ability = nil)
-      self.filter_params = filter_params
-      self.ability = ability
+    def initialize(filter_params, ability, order_params)
```

Comment:
we should remove the word `params` from the params :)

`def initialize(filter, ability, order)`